### PR TITLE
A complete rework of the skeleton DSL

### PR DESCRIPTION
**Pull request by warmwaffles**
This was done in order to provide a better look and feel to the skeleton
form. As time went on at KISSmetrics, I realized we had deeper needs
than what was initially required.

An Options serializer will be implemented soon to allow for options
requests to spit out only the required information for a specific end
point. It will be Swagger-esque but a subset of what is defined. Only
the required models will be included along with omission of the
information

This is still a work in progress.

### CODE REVIEW

File: lib/skeleton/license.rb [hunk 1]
```diff
@@ -1,16 +1,5 @@
-require 'skeleton/model'
-
 module Skeleton
-  class License < Model
+  class License
```
Comment (healyk) [inline lib/skeleton/license.rb:2]: Do you need a whole class for a License if it is just 2 fields?

Comment (nateklaiber) [inline lib/skeleton/license.rb:2]: @healyk - what would you recommend?

Comment (healyk) [inline lib/skeleton/license.rb:2]: If you don't mind validating it somewhere else a hash works fine.

If you want to keep a strict set of variables with names a Struct might be better.

``` ruby
License = Struct.new(:url, :name)
```

Comment (warmwaffles) [inline lib/skeleton/license.rb:2]: For some reason I enjoy boiler plate code. I could just have `class License < Struct.new(:url, :name)` to make sure it still is a class.

**Comment by nateklaiber** [comment]
:shipit:

**Pull request closed without merging**